### ISSUE TITLE
`tsc` complains `require()` file path outside of rootDir only when `await import()` is used

### ISSUE THREAD

**Issue by ychi** [microsoft/TypeScript#38611]
**TypeScript Version:**  4.0.0-dev.20200516

**Search Terms:** commonjs, esmodule, rootDir

**Code**
Consider this file:
```ts
onst {version: VERSION} = require('../package.json');
console.log(VERSION);

async function foo() {
    let obj = {};
    console.log(obj);
}

export async function bar() {
    console.log('foo');
    await foo();
}
```
And this file:
```ts
const {version: VERSION} = require('../package.json');
console.log(VERSION);

async function foo() {
    let obj = await import('./anotherFile');
    console.log(obj);
}

export async function bar() {
    console.log('foo');
    await foo();
}
```

**Expected behavior:**
Running `tsc` to compile them, they should both succeed or both fail?
**Actual behavior:**
The first file can be compiled smoothly, while the second was complained about `error TS6059` and `error TS6307`.
**Playground Link:** https://glitch.com/~aromatic-bittersweet-biplane

**Related Issues:**

**Comment by cspotcode** [comment]
I am seeing this bug triggered by a slightly different reason: adding an `import()` type annotation.

I have created a minimal reproduction here: https://github.com/TypeStrong/ts-node-repros/tree/ts-38611
It executed here: https://github.com/TypeStrong/ts-node-repros/runs/682702614

**Comment by sheetalkamat** [comment]
@ychi your repro link seems to be not working, Can you please share the repro again. Thanks

**Comment by sheetalkamat** [comment]
Actually @cspotcode repro seems to be working.. Thanks.
